client/src/endpoint/trades.rs

```diff
@@ -0,0 +1,257 @@
+    /// use stellar_client::sync::Client;
+    /// use stellar_client::endpoint::trades;
+    ///
+    /// let client      = Client::horizon_test().unwrap();
+    /// let endpoint    = trades::All::default().limit(3);
+    /// let records     = client.request(endpoint).unwrap();
+    /// #
+    /// # assert_eq!(records.records().len(), 3);
+    /// ```
+    pub fn limit(mut self, limit: u32) -> Self {
+        self.limit = Some(limit);
+        self
+    }
+
+    fn has_query(&self) -> bool {
+        self.order.is_some() || self.cursor.is_some() || self.limit.is_some() ||
+        self.offer_id.is_some() || self.base_asset.is_some() ||
+        self.counter_asset.is_some()
+    }
+}
+
+impl IntoRequest for All {
+    type Response = Records<Trade>;
+
+    fn into_request(self, host: &str) -> Result<Request<Body>> {
+        let mut uri = format!("{}/trades", host);
+
+        if self.has_query() {
+            uri.push_str("?");
+
+            if let Some(base_asset) = self.base_asset {
+                if base_asset.is_native() {
+                    uri.push_str(&format!("base_asset_type=native&"));
+                } else {
+                    uri.push_str(&format!("base_asset_type={}&base_asset_code={}&base_asset_issuer={}&", 
+                    base_asset.asset_type(), base_asset.code(), base_asset.issuer()));
+                }  
+            }
+
+            if let Some(counter_asset) = self.counter_asset {
+                if counter_asset.is_native() {
+                    uri.push_str(&format!("counter_asset_type=native&"));
+                } else {
+                    uri.push_str(&format!("counter_asset_type={}&counter_asset_code={}&counter_asset_issuer={}&", 
+                    counter_asset.asset_type(), counter_asset.code(), counter_asset.issuer()));
+                }
+            }
+
+            if let Some(offer_id) = self.offer_id {
+                uri.push_str(&format!("offer_id={}&", offer_id));
+            }
+
+            if let Some(order) = self.order {
+                uri.push_str(&format!("order={}&", order.to_param()));
+            }
+
+            if let Some(cursor) = self.cursor {
+                uri.push_str(&format!("cursor={}&", cursor));
+            }
+
+            if let Some(limit) = self.limit {
+                uri.push_str(&format!("limit={}", limit));
+            }
+        }
+
+        let uri = Uri::from_str(&uri)?;
+        let request = Request::get(uri).body(Body::None)?;
+        Ok(request)
+    }
+}
+
+#[cfg(test)]
+mod all_trades_tests {
+    use super::*;
+
+    #[test]
+    fn it_leaves_off_the_params_if_not_specified() {
+        let ep = All::default();
+        let req = ep.into_request("https://www.google.com").unwrap();
+        assert_eq!(req.uri().path(), "/trades");
+        assert_eq!(req.uri().query(), None);
+    }
+
+    #[test]
+    fn it_puts_the_query_params_on_the_uri() {
+        let ep = All::default()
+            .base_asset(AssetIdentifier::native())
```

Comment:
I think that we should have at least two tests here. I'd like to see us test both the native and alpha's for each asset.